### DISCUSSION TITLE
Event Listener for Specific Data Object Classes

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16928] [Q&A / Support] [answered]
Is there any way to define an `event_listener` for _specific_ data object classes?

Right now I have:
```
{ name: kernel.event_listener, event: pimcore.dataobject.preUpdate, method: onPreUpdate }
```
and then I need to do an `instanceof` check in the method:
```php
public function onPreUpdate(DataObjectEvent $e){
  if ($e->getObject() instanceof DataObject\Product){
   // ...
  }
}
```
however for for Workflows I am able to do:
```
{ name: kernel.event_listener, event: workflow.product_workflow.transition.approved, method: onApproved }
```
where I can subscribe to a specific workflow.

**Comment by markus-moser** [comment, marked as answer]
There are no data object class specific events so you need to do it with the instanceof approach. For workflows there are more fine-grained events as it is based on the Symfony workflow engine which offers  these kind of events.
